ui/src/shared/utils/resourceToTemplate.test.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  labelToRelationship,
+  labelToIncluded,
+  taskToTemplate,
+} from 'src/shared/utils/resourceToTemplate'
+import {TemplateType, Task as TaskApi} from '@influxdata/influx'
+import {Label, Task} from 'src/types/v2'
+
+const myfavelabel: Label = {
+  id: '1',
+  name: '1label',
+  properties: {color: 'fffff', description: 'omg'},
+}
+
+const myfavetask: Task = {
+  authorizationID: '037b084ed9abc000',
+  every: '24h0m0s',
+  flux:
+    'option task = {name: "lala", every: 86400000000000ns, offset: 60000000000ns}\n\nfrom(bucket: "defnuck")\n\t|> range(start: -task.every)',
+  id: '037b0877b359a000',
+  labels: [
+    {
+      id: '037b0c86a92a2000',
+      name: 'yum',
+      properties: {color: '#FF8564', description: ''},
+    },
+  ],
+  name: 'lala',
+  offset: '1m0s',
+  org: 'org',
+  orgID: '037b084ec8ebc000',
+  status: TaskApi.StatusEnum.Active,
+}
+
+describe('resourceToTemplate', () => {
+  describe('labelToRelationship', () => {
+    it('converts a label to a relationship struct', () => {
+      const actual = labelToRelationship(myfavelabel)
+      const expected = {type: TemplateType.Label, id: myfavelabel.id}
+
+      expect(actual).toEqual(expected)
+    })
+  })
+  describe('labelToIncluded', () => {
+    it('converts a label to a data structure in included', () => {
+      const actual = labelToIncluded(myfavelabel)
+      const expected = {
+        type: TemplateType.Label,
+        id: myfavelabel.id,
+        attributes: {
+          name: myfavelabel.name,
+          properties: {
+            color: myfavelabel.properties.color,
+            description: myfavelabel.properties.description,
+          },
+        },
+      }
+
+      expect(actual).toEqual(expected)
+    })
+  }),
+    describe('taskToTemplate', () => {
+      it('converts a task to a template', () => {
+        const actual = taskToTemplate(myfavetask)
+        const expected = {
+          content: {
+            data: {
```

Comment:
seems like `data` and `included` could be flattened to a single level under `content` or something... or is there a meaningful distinction between `data` and `included`? what's the extra level of nesting under `content` for? I think I just lack context, but am curious

Reply:
yeah, so template meta, labels, and id have a schema, but everything under content is schema-less, data refers to the primary resource, and included is resources related to primary resource. so if this is a task template, data contains task, included contains labels, etc.